packages/next/src/client/components/react-dev-overlay/app/hot-reloader-client.tsx

```diff
@@ -554,6 +562,15 @@ export default function HotReload({
     }
   }, [dispatch])
 
+  //  We render a separate error overlay at the root when an error is thrown from rendering RSC, so
+  //  we should not render an additional error overlay in the descendent. However, we need to
+  //  keep rendering these hooks to ensure HMR works when the error is addressed.
+  const _isErrorThrownWhileRenderingRsc = useSyncExternalStore(
```

Comment:
sounds we could rename `_isErrorThrownWhileRenderingRsc` to `hasDefaultDevOverlay` or sth indicating there's an error overlay, so we don't wrap it.

Reply:
Changed to `shouldRenderRootLevelErrorOverlay`